File: WasnieApi/src/Wasnie.Domain/Authorization/TierLimits.cs
namespace Wasnie.Domain.Authorization;

public sealed record TierLimitValues(int MaxPayees, int MaxPlans);

public static class TierLimits
{
    public static readonly IReadOnlyDictionary<Tier, TierLimitValues> Limits =
        new Dictionary<Tier, TierLimitValues>
        {
            [Tier.Free]       = new(MaxPayees: 5,           MaxPlans: 1),
            [Tier.Starter]    = new(MaxPayees: 25,          MaxPlans: 5),
            [Tier.Growth]     = new(MaxPayees: 75,          MaxPlans: 15),
            [Tier.Scale]      = new(MaxPayees: 150,         MaxPlans: int.MaxValue),
            [Tier.Enterprise] = new(MaxPayees: int.MaxValue, MaxPlans: int.MaxValue),
        };
}

File: WasnieUi/src/app/shared/services/tier-limits.ts
export interface TierLimitConfig {
  maxPlans: number;   // -1 = unlimited
  maxPayees: number;  // -1 = unlimited
}

export const TIER_LIMITS: Record<string, TierLimitConfig> = {
  Free:       { maxPlans: 1,  maxPayees: 5   },
  Starter:    { maxPlans: 5,  maxPayees: 25  },
  Growth:     { maxPlans: 15, maxPayees: 75  },
  Scale:      { maxPlans: -1, maxPayees: 150 },
  Enterprise: { maxPlans: -1, maxPayees: -1  },
};
